### PR TITLE
Add eks managed nodegroup submodule

### PR DESCRIPTION
**Pull request by niroz89**
- Add managed node group module
- Add managed node group ASG tags for scale from 0
- Bump terraform EKS upstream module for root and submodules
- Update README

Sample tags created for BT cluster eks managed ASG.
```hcl
autoscaling_group_tags = {
  "cicd-subnet-14868760" = {
    "k8s.io/cluster-autoscaler/node-template/label/aws.amazon.com/spot" = true
    "k8s.io/cluster-autoscaler/node-template/label/lifecycle" = "spot"
    "k8s.io/cluster-autoscaler/node-template/label/node.kubernetes.io/instance-type" = "m5a.large"
    "k8s.io/cluster-autoscaler/node-template/label/topology.ebs.csi.aws.com/zone" = "ap-southeast-1b"
    "k8s.io/cluster-autoscaler/node-template/label/topology.kubernetes.io/zone" = "ap-southeast-1b"
  }
  "cicd-subnet-c5ef0fa0" = {
    "k8s.io/cluster-autoscaler/node-template/label/aws.amazon.com/spot" = true
    "k8s.io/cluster-autoscaler/node-template/label/lifecycle" = "spot"
    "k8s.io/cluster-autoscaler/node-template/label/node.kubernetes.io/instance-type" = "m5a.large"
    "k8s.io/cluster-autoscaler/node-template/label/topology.ebs.csi.aws.com/zone" = "ap-southeast-1a"
    "k8s.io/cluster-autoscaler/node-template/label/topology.kubernetes.io/zone" = "ap-southeast-1a"
  }
}
```
image

To Do 
- [ ]  make use of `autoscaling_group_tags` for self managed node groups - separate PR.

### CODE REVIEW

File: modules/eks_managed_nodes/asg_tags.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  cluster_autoscaler_asg_tags = {
+    for name, group in local.eks_managed_node_groups : name => merge(
+      try(local.cluster_autoscaler_label_tags[name], {}),
+      try(local.cluster_autoscaler_taint_tags[name], {}),
+      local.cluster_autoscaler_implicit_tags[name],
+    )
+  }
+}
+
```
Comment (lawliet89) [inline modules/eks_managed_nodes/asg_tags.tf:46]: Minor suggestion to try and reduce the use of `try` in `local.cluster_autoscaler_asg_tags`.

In `local.cluster_autoscaler_label_tags` and `local.cluster_autoscaler_taint_tags`, you can remove the `if` clause in the `for` expression to always produce a value for each group. (And remove the `merge`)

```hcl
  cluster_autoscaler_label_tags = {
    for name, group in local.eks_managed_node_groups : name => {
        for label_name, label_value in try(group.labels, {}) : "k8s.io/cluster-autoscaler/node-template/label/${label_name}" => label_value
    }
  }
```

Might be a syntax error for the above in terms of line breaks. Need to tweak the formatting a bit.

File: modules/eks_managed_nodes/variables.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+variable "eks_managed_node_group_defaults" {
+  description = "Map of EKS managed node group default configurations"
+  type        = any
+  default = {
+    metadata_options = {
+      http_endpoint               = "enabled"
+      http_tokens                 = "required"
+      http_put_response_hop_limit = 1
+      instance_metadata_tags      = "disabled"
+    }
```
Comment (lawliet89) [inline modules/eks_managed_nodes/variables.tf:56]: These kind of "conflict" with `var.force_imdsv2` and `var.force_irsa` right? Maybe we should just omit this from the defaults?

Comment (lawliet89) [reply modules/eks_managed_nodes/variables.tf:56]: I realise that this is also in the `self_managed_node_groups` variable. I guess we might want to remove it from there too.

**Review by lawliet89 — COMMENTED** [review]
Thank you @niroz89 for the very nice PR. Some minor comments but nothing major.

File: modules/eks_managed_nodes/asg_tags.tf (hunk 1 above)
Comment (niroz89) [reply modules/eks_managed_nodes/asg_tags.tf]: Thank you @lawliet89 . Done.

File: modules/eks_managed_nodes/variables.tf (hunk 2 above)
Comment (niroz89) [reply modules/eks_managed_nodes/variables.tf]: Done.

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**